packages/test/test-end-to-end-tests/src/test/summaries.spec.ts

```diff
@@ -307,6 +325,73 @@ describeNoCompat("Summaries", (getTestObjectProvider) => {
 		);
 	});
 
+	itExpects(
+		"full initialization of data object should not happen by default",
+		[
+			{
+				eventName: "fluid:telemetry:Container:Request_cancel",
+				error: "Non interactive/summarizer client's data object should not be initialized",
+			},
+		],
+		async () => {
+			const TestDataObjectType1 = "@fluid-example/test-dataStore1";
+			const dataStoreFactory1 = new DataObjectFactory(
+				TestDataObjectType1,
+				TestDataObject1,
+				[],
+				[],
+			);
+			const registryStoreEntries = new Map<string, Promise<IFluidDataStoreFactory>>([
+				[dataStoreFactory1.type, Promise.resolve(dataStoreFactory1)],
+			]);
+			const runtimeFactory = new ContainerRuntimeFactoryWithDefaultDataStore(
+				dataStoreFactory1,
+				registryStoreEntries,
+				undefined,
+				[],
+			);
+
+			// Create a container for the first client.
+			const container1 = await provider.createContainer(runtimeFactory);
+			await assert.doesNotReject(
+				requestFluidObject<TestDataObject1>(container1, "/"),
+				"Initial creation of container and data store should succeed.",
+			);
+
+			// Create a summarizer for the container and do a summary shouldn't throw.
+			const createSummarizerResult = await createSummarizerFromFactory(
+				provider,
+				container1,
+				dataStoreFactory1,
+				undefined,
+				ContainerRuntimeFactoryWithDefaultDataStore,
+				registryStoreEntries,
+			);
+			const summarizer = createSummarizerResult.summarizer;
+			assert.doesNotThrow(
+				async () => summarizeNow(summarizer, "test"),
+				"Summarizing should not throw",
+			);
```

Comment:
`summarizeNow` is itself an async function so if it's not awaited it won't immediately throw. I think this want to read:
```suggestion
			await assert.doesNotReject(
				summarizeNow(summarizer, "test"),
				"Summarizing should not throw",
			);
```
Since `await assert.doesNotReject()` will await the Promise passed to it in its first argument.

Reply:
@chentong7 It's a good idea to manually validate that what the test is validating is correct. For example, before adding the assert here, I would store the summarizeNow result in a variable, add a breakpoint here and check what the contents of the variable. That would immediately tell you that the result is just a promise so it would never throw. Relying solely on asserts is never really a good idea - they should be added after you have manualy validated your expectation.